main.c

```diff
@@ -23,6 +23,33 @@
 #include <openssl/bio.h>
 #include <openssl/x509.h>
 
+#if OPENSSL_VERSION_NUMBER >= 0x1010103fL
```

Comment:
```
{
  node: '12.7.0',
  v8: '7.5.288.22-node.16',
  uv: '1.30.1',
  zlib: '1.2.11',
  brotli: '1.0.7',
  ares: '1.15.0',
  modules: '72',
  nghttp2: '1.39.1',
  napi: '4',
  llhttp: '1.1.4',
  http_parser: '2.8.0',
  openssl: '1.1.1c',
  cldr: '35.1',
  icu: '64.2',
  tz: '2019a',
  unicode: '12.1'
}

```

Reply:
I don't think this if statement is entirely correct no I should really check it works for each version of node but i haven't dont that

Reply:
i'll change this to 0x10101000L so that should mean anything above openssl 1.1.0 should use this code

i've tested it on node8 and 12. from the docs it looks like all supported version of node use the same version

> All supported versions of Node.js use OpenSSL v1.1.1, therefore all active release lines are impacted by this update: v10.x, v12.x, v13.x, and v14.x ( 14.0.0 is to be released on the 21st of April, by coincidence).

https://nodejs.org/en/blog/vulnerability/april-2020-openssl-updates/

so i think we're good
node 8 uses openssl 1.0.2s which translates to # define OPENSSL_VERSION_NUMBER  0x1000213fL
so for node 8 this if statement will be false which is correct

Reply:
that sounds good then